keanu-project/src/test/java/io/improbable/keanu/vertices/dbltensor/probabilistic/TensorGammaVertexTest.java

```diff
@@ -0,0 +1,232 @@
+package io.improbable.keanu.vertices.dbltensor.probabilistic;
+
+import io.improbable.keanu.distributions.continuous.Gamma;
+import io.improbable.keanu.vertices.dbl.probabilistic.GammaVertex;
+import io.improbable.keanu.vertices.dbltensor.*;
+import io.improbable.keanu.vertices.dbltensor.nonprobabilistic.ConstantTensorVertex;
+import io.improbable.keanu.vertices.dbltensor.nonprobabilistic.diff.TensorPartialDerivatives;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.function.Supplier;
+
+import static io.improbable.keanu.vertices.dbltensor.probabilistic.ProbabilisticDoubleTensorContract.moveAlongDistributionAndTestGradientOnARangeOfHyperParameterValues;
+import static org.junit.Assert.assertEquals;
+
+public class TensorGammaVertexTest {
+
+    private KeanuRandom random;
+
+    private static final double DELTA = 0.0001;
+
+    @Before
+    public void setup() {
+        random = new KeanuRandom(1);
+    }
+
+    @Test
+    public void matchesKnownLogDensityOfScalar() {
+        GammaVertex gamma = new GammaVertex(0.5, 1, 1.5, new Random(1));
```

Comment:
The recent PR for random sets us up to only use KeanuRandom and never passes a random to a vertex constructor. That might clean this up a bit.

Reply:
Nice I'll merge it in